python/pyspark/sql/functions/builtin.py

```diff
@@ -10985,7 +10994,9 @@ def split(str: "ColumnOrName", pattern: str, limit: int = -1) -> Column:
     >>> df.select(split(df.s, '[ABC]', -1).alias('s')).collect()
     [Row(s=['one', 'two', 'three', ''])]
     """
-    return _invoke_function("split", _to_java_column(str), pattern, limit)
+    pattern = pattern if isinstance(pattern, Column) else lit(pattern)
+    limit = lit(limit) if isinstance(limit, int) else limit
```

Comment:
I think `lit` function accept both Column and int

Reply:
`limit` can also be of type `str` referring to a column, hence the check to avoid making the column name a string literal.

Reply:
yea, we can use it for `pattern`